src/Microsoft.AspNetCore.Http.Connections.Client/Internal/HttpOptionsMessageHandler.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Connections.Client.Internal
+{
+    public class HttpOptionsMessageHandler : DelegatingHandler
+    {
+        private readonly HttpOptions _httpOptions;
+
+        public HttpOptionsMessageHandler(HttpMessageHandler inner, HttpOptions options) : base(inner)
+        {
+            _httpOptions = options;
+        }
+
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (_httpOptions?.Headers != null)
```

Comment:
I'd like a comment explaining why these headers are set in a message handler instead of on the `HttpRequestMessage` prior to using `HttpClient`.

Reply:
I don't understand. Are you asking for a comment that says we're using a message handler so we don't have to manually set common headers for every request?

Actually thinking about it now, I wonder if it makes sense to set DefaultHeaders on the http client itself

Reply:
I asked for a comment because I was curious why we were changing from the `PrepareRequst` to using a handler. Is it just a refactor or does it fix an issue? If it fixes an issue then it would be good to say why we're doing something in such a way.

Reply:
Oh, this is just a better way to execute code on *all* requests. We don't have to sprinkle code around the code base. Just like you did with the LoggingMessageHandler.